### PR TITLE
Resource tabular list & sortable table rework

### LINKED ISSUE #57: Implement tablular catalogue item listing
Items in a CatPrez catalogue need to be listed in a tabular manner, as vocabs now are, rather than in a flat list.

e.g. https://data.idnau.org/c/catalogs/pid:democat should look like https://data.idnau.org/v/vocab with perhaps Title, Publisher, Creator and Created Date as the table properties

### PR DESCRIPTION
**Pull request by jamiefeiss**
Added a tabular listing of Resources on a Catalog page, same as the Vocab listing page. Both the list of Resources and listing VocPrez items now share a more generic component: `SortableTabularList.vue`.

This PR also removes `VocabItemList.vue` & relevant types as there are no longer used.

Resolves #57.

image

### CODE REVIEW

File: src/views/PropTableView.vue [hunk 1]
```diff
@@ -250,7 +256,16 @@ function getChildren() {
...
-                }
+                } else if (item.value.type === qname("dcat:Catalog") && q.predicate.value === qname("dcterms:publisher")) {
+                    const publisher: ListItemSortable = { label: q.object.value };
+                    child.extras.publisher = publisher;
+                } else if (item.value.type === qname("dcat:Catalog") && q.predicate.value === qname("dcterms:creator")) {
+                    const creator: ListItemSortable = { label: q.object.value };
+                    child.extras.creator = creator;
+                } else if (item.value.type === qname("dcat:Catalog") && q.predicate.value === qname("dcterms:issued")) {
+                    const publisher: ListItemSortable = { label: q.object.value };
+                    child.extras.publisher = publisher;
```
Comment (edmondchuc) [inline src/views/PropTableView.vue:267]: This should be assigned to `child.extras.issued` instead of `publisher`.

**Review by edmondchuc — CHANGES_REQUESTED** [review]

File: src/views/PropTableView.vue (hunk 1 above)
Comment (jamiefeiss) [reply src/views/PropTableView.vue]: Ah whoops, good catch. Fixed in latest commit

File: src/views/PropTableView.vue [hunk 2]
```diff
@@ -250,7 +256,16 @@ function getChildren() {
                     child.link = q.object.value;
                 } else if (q.predicate.value === qname("a")) {
                     child.type = q.object.value;
-                }
+                } else if (item.value.type === qname("dcat:Catalog") && q.predicate.value === qname("dcterms:publisher")) {
+                    const publisher: ListItemSortable = { label: q.object.value };
+                    child.extras.publisher = publisher;
+                } else if (item.value.type === qname("dcat:Catalog") && q.predicate.value === qname("dcterms:creator")) {
+                    const creator: ListItemSortable = { label: q.object.value };
+                    child.extras.creator = creator;
```
Comment (edmondchuc) [inline src/views/PropTableView.vue:264]: Are we able to retrieve the label of the object here as well and store it as an object with `iri` and `label`? I think it should display as a hyperlink in the same way as how the vocab listing table shows the reg status and the derivation mode.

**Review by edmondchuc — CHANGES_REQUESTED** [review]

File: src/views/PropTableView.vue (hunk 2 above)
Comment (jamiefeiss) [reply src/views/PropTableView.vue]: Added in latest commit

**Review by edmondchuc — APPROVED** [review]
Thanks, looks great!

**Pull request merged**